djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor_plugins/cmsplugins/plugin.js

```diff
@@ -404,36 +404,40 @@
                 if (!that.options.delete_on_cancel && !that.unsaved_child_plugins.length) {
                     return;
                 }
-                if (that.unsaved_child_plugins.length) {
-                    e.preventDefault();
-                    CMS.API.Toolbar.showLoader();
-                    var data = {
-                        token: that.options.action_token
-                    };
-
-                    if (!that.options.delete_on_cancel) {
```

Comment:
The logic here seems wrong in the first place. `delete_on_cancel` should be `True` if we intend to delete on cancel. With the `!` we check the opposite. So this has been changed further down.

Reply:
`delete_on_cancel` is false if the TextPlugin is being created. this whole thing covered the case when the plugin didn't yet exist but we are creating the TextPlugin and immediately adding children to it.

when `delete_on_cancel` is true - that means that the TextPlugin already exists and we need to handle cancelling the children that were created during the current editing

Reply:
(or at least that was the idea, i haven't seen this code in a while)